torchinductor/lowering.py

```diff
@@ -1538,6 +1537,107 @@ def fn(idx, return_index):
     return r1, r2
 
 
+@register_lowering(aten.max_pool2d_with_indices_backward, type_promote=False)
+def max_pool2d_with_indices_backward(
+    grad_output, x, kernel_size, stride, padding, dilation, ceil_mode, indices
+):
+    assert dilation == 1 or all(d == 1 for d in dilation)
+    assert isinstance(x, TensorBox)
+    assert len(kernel_size) == 2
+    assert len(stride) == 2
+    assert padding == 0 or len(padding) == 2
+    assert len(x.get_size()) in (3, 4)
```

Comment:
Do we need to add memory format checking here?

Reply:
This should work for all memory formats.  This 3/4 check is actually just for omitting the batch dim.